src/features/auth/authService.ts

```diff
@@ -1,45 +1,88 @@
-import { UserManager, User, UserManagerSettings, Log } from 'oidc-client';
+import { UserManager, Log, WebStorageStateStore } from 'oidc-client-ts';
+import type { User, UserManagerSettings } from 'oidc-client-ts';
 import axios from 'axios';
+import * as Sentry from '@sentry/browser';
+import AppConfig from '../../app/AppConfig';
 
 const origin = window.location.origin;
 export const API_TOKENS = 'apiTokens';
+export const REFRESH_TOKEN = 'refreshToken';
+
+export type ApiTokenClientProps = {
+  url: string;
+  queryProps?: { permission: string; grant_type: string };
+  audiences?: string[];
+};
 
 class AuthService {
-  private userManager: UserManager;
+  private readonly userManager: UserManager;
+  private readonly apiTokensClientConfig: ApiTokenClientProps;
+  private readonly authServerType: 'KEYCLOAK' | 'TUNNISTAMO';
+  private readonly audiences: string[];
 
   constructor() {
-    const {
-      REACT_APP_TUNNISTAMO_SCOPE_BERTHS: berthScope,
-      REACT_APP_TUNNISTAMO_SCOPE_PROFILE: profileScope,
-    } = process.env;
+    this.authServerType = AppConfig.oidcServerType;
+    this.audiences = AppConfig.oidcAudiences ?? [AppConfig.oidcBerthApiClientId];
+
     const settings: UserManagerSettings = {
-      /* eslint-disable @typescript-eslint/camelcase */
-      authority: process.env.REACT_APP_TUNNISTAMO_URI,
-      automaticSilentRenew: true,
-      client_id: process.env.REACT_APP_TUNNISTAMO_CLIENT_ID,
+      loadUserInfo: true,
+      userStore: new WebStorageStateStore({ store: window.localStorage }),
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      response_type: AppConfig.oidcReturnType,
+      authority: AppConfig.oidcAuthority,
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      client_id: AppConfig.oidcClientId,
+      scope: AppConfig.oidcScope,
+      // eslint-disable-next-line @typescript-eslint/camelcase
       redirect_uri: `${origin}/callback`,
-      post_logout_redirect_uri: origin,
-      response_type: 'id_token token',
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      post_logout_redirect_uri: `${origin}/logout`,
+      automaticSilentRenew: AppConfig.oidcAutomaticSilentRenew,
+      checkSessionIntervalInSeconds: AppConfig.oidcSessionPollerIntervalInMs,
```

Comment:
Is this correct? The left variable name points to seconds, the right to milliseconds?

Reply:
I'm wondering, that shouldn't this be less than the expiration time set in the JWT?

Reply:
https://github.com/City-of-Helsinki/kukkuu-admin/pull/325